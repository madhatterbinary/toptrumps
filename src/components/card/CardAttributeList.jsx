import React from 'react';
import PropTypes from 'prop-types';
import CardAttributeRow from './CardAttributeRow';
import attributes from '../../constants/attributes';

const CardAttributeList = (props) => {
  const {
    card,
    handleAttributeSelection,
    status,
    selectedAttribute,
  } = props;

  const attributeListItems = Object.keys(attributes).map((attributeKey) => {
    return (
    <CardAttributeRow
      key={`attribute-${attributeKey}`}
      attributeKey={attributeKey}
      label={attributes[attributeKey]}
      value={card[attributeKey] || ""}
      handleAttributeClick={handleAttributeSelection}
      selectedAttribute={selectedAttribute}
      status={status}
    />
    );
  });

  return (
    <div className="card-attribute-list">
      {attributeListItems}
    </div>
  );
}

CardAttributeList.propTypes = {
  card: PropTypes.object.isRequired,
  handleAttributeSelection: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
  selectedAttribute: PropTypes.string
}

export default CardAttributeList;
