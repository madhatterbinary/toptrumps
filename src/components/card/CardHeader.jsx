import React from 'react';
import PropTypes from 'prop-types';

const CardHeader = (props) => {
  const {
    title,
    model,
    manufacturer,
  } = props;

  return (
    <div className="card-header">
      <h3>{title}</h3>
      <div className="card-additional-info">
        <div>Model: {model}</div>
        <div>Manufacturer: {manufacturer}</div>
      </div>
    </div>
  );
}

CardHeader.propTypes = {
  title: PropTypes.string,
  model: PropTypes.string,
  manufacturer: PropTypes.string,
}

export default CardHeader;
