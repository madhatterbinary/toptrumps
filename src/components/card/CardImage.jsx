import React from 'react';
import PropTypes from 'prop-types';
import images from '../../images/index';

const CardImage = (props) => (
  <div className="card-image-container">
    <img src={images[props.image]} className="card-image" alt="" />
  </div>
);

CardImage.propTypes = {
  image: PropTypes.string,
}

export default CardImage;
