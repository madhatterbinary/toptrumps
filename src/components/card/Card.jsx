import React from 'react';
import PropTypes from 'prop-types';
import CardHeader from './CardHeader';
import CardImage from './CardImage';
import CardAttributeList from './CardAttributeList';

const Card = (props) => {
  const {
    isClosed,
    card,
    handleAttributeSelection,
    status,
    selectedAttribute,
    isWinner
  } = props;

  return (
    <div className={`card ${isClosed && 'closed'}`}>
      {!isClosed ? (
        <div className="card-body">
          <CardHeader
            title={card.name}
            model={card.model}
            manufacturer={card.manufacturer}
          />
          <CardImage
            image={card.image}
          />
          <CardAttributeList
            card={card}
            handleAttributeSelection={handleAttributeSelection}
            status={status}
            selectedAttribute={selectedAttribute || ""}
            isWinner={isWinner}
          />
        </div>
      ) : null}
    </div>
  );
}

Card.propTypes = {
  isClosed: PropTypes.bool.isRequired,
  card: PropTypes.object.isRequired,
  handleAttributeSelection: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
}

export default Card;
