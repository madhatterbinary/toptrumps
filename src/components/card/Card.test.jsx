import React from 'react';
import Card from './Card';
import CardHeader from './CardHeader';
import CardImage from './CardImage';
import CardAttributeList from './CardAttributeList';
import TestRenderer from 'react-test-renderer';
 
jest.mock('./CardHeader', () => () => 'CardHeader');
jest.mock('./CardImage', () => () => 'CardImage');
jest.mock('./CardAttributeList', () => () => 'CardAttributeList');

describe('Card', () => {
  it('renders', () => {
    const component = TestRenderer.create(<Card isClosed={true} card={{}} handleAttributeSelection={() =>{}} status="" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(component.root.findByProps({className: "card closed"}).children).toEqual([]);
       expect(component.root.findAllByType(CardHeader)).not.toBeNull();
       expect(component.root.findAllByType(CardImage)).not.toBeNull();
       expect(component.root.findAllByType(CardAttributeList)).not.toBeNull();
  });
});
