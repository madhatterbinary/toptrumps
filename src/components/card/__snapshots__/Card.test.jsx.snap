// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `<div />`;

exports[`Card renders 1`] = `
<div
  className="card closed"
/>
`;
